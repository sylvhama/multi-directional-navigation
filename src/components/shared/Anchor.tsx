import styled from "styled-components";

export const Anchor = styled.a`
  color: #03a9f4;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;
