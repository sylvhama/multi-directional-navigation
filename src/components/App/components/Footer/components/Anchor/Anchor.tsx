import styled from "styled-components";
import { IsFocusedProps } from "../../../../../shared/types";

export const Anchor = styled.a<IsFocusedProps>`
  position: relative;
  color: #03a9f4;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }

  ${({ isFocused }) =>
    isFocused &&
    `
      &::before{
        position: absolute;
        top: 50%;
        left: -1.5rem;
        content: 'ðŸ”—';
        margin-top: -0.5rem;
        font-size: 1rem;
        color: white;
        line-height: 1;
      }
    `}
`;
