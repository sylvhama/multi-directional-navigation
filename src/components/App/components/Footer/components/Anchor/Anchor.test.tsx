import "jest-styled-components";
import React from "react";
import { render } from "@testing-library/react";

import { Anchor } from "./Anchor";

describe.each([true, false])("<Anchor />", (isFocused) => {
  it(`renders style when isFocused is ${isFocused}`, () => {
    const { container } = render(<Anchor isFocused={isFocused} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
