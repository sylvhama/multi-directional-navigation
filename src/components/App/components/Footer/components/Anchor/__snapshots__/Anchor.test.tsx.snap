// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> renders style when isFocused is false 1`] = `
.c0 {
  position: relative;
  color: #03a9f4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<a
  class="c0"
/>
`;

exports[`<Anchor /> renders style when isFocused is true 1`] = `
.c0 {
  position: relative;
  color: #03a9f4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0::before {
  position: absolute;
  top: 50%;
  left: -1.5rem;
  content: 'ðŸ”—';
  margin-top: -0.5rem;
  font-size: 1rem;
  color: white;
  line-height: 1;
}

<a
  class="c0"
/>
`;
